### PR TITLE
Updated to latest Swift 3 Preview 2 Syntax

### PR DESCRIPTION
**Pull request by harlanhaskins**

### CODE REVIEW

File: Sources/Utility/File.swift [hunk 1]
```diff
@@ -21,11 +21,11 @@ public func fopen(_ path: String, mode: FopenMode = .read) throws -> NSFileHandl
...
-            guard NSFileManager.`default`().createFile(atPath: path, contents: nil) else {
+            guard NSFileManager.default.createFile(atPath: path, contents: nil) else {
                 throw Error.couldNotCreateFile(path: path)
             }
         #else
-            guard NSFileManager.defaultManager().createFile(atPath: path, contents: nil) else {
+            guard NSFileManager.default.createFile(atPath: path, contents: nil) else {
                 throw Error.couldNotCreateFile(path: path)
             }
         #endif
```
Comment (aciidb0mb3r) [inline Sources/Utility/File.swift]: yay conditional can be removed now!

Comment (harlanhaskins) [inline Sources/Utility/File.swift]: 😎

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Comment by harlanhaskins** [comment]
It looks...like the test didn't do anything?

**Comment by shahmishal** [comment]
@swift-ci please test

**Comment by shahmishal** [comment]
Linux failure: 

```
/home/buildnode/jenkins/workspace/swift-package-manager-PR-Linux/swiftpm/Sources/Utility/misc.swift:33:41: error: cannot call value of non-function type 'NSFileManager'
            return NSFileManager.default()
                   ~~~~~~~~~~~~~~~~~~~~~^
```

OS X failure: 

```
/Users/buildnode/jenkins/workspace/swift-package-manager-PR-osx/swiftpm/Sources/Utility/Git.swift:70:106: error: method 'processInfo' was used as a property; add () to call it
                try system(Git.tool, "-C", path, "fetch", "--tags", "origin", environment: NSProcessInfo.processInfo.environment, message: nil)
                                                                                                         ^
                                                                                                                    ()
/Users/buildnode/jenkins/workspace/swift-package-manager-PR-osx/swiftpm/Sources/Utility/File.swift:23:29: error: method 'default' was used as a property; add () to call it
        guard NSFileManager.default.createFile(atPath: path, contents: nil) else {
                            ^
                                   ()
/Users/buildnode/jenkins/workspace/swift-package-manager-PR-osx/swiftpm/Sources/Utility/makeDirectories.swift:20:23: error: method 'default' was used as a property; add () to call it
    try NSFileManager.default.createDirectory(atPath: path, withIntermediateDirectories: true, attributes: [:])
                      ^
                             ()
/Users/buildnode/jenkins/workspace/swift-package-manager-PR-osx/swiftpm/Sources/Utility/removeFileTree.swift:15:23: error: method 'default' was used as a property; add () to call it
    try NSFileManager.default.removeItem(atPath: path)
                      ^
                             ()
<unknown>:0: error: build had 1 command failures
```

**Comment by abertelrud** [comment]
It looks like this patch still uses NS prefixes.  With the shims gone (47f12a2eb0767a44160757133cba0fe58a8c5176), it doesn't seem to me that that will still build if merged into master.

**Comment by harlanhaskins** [comment]
Hmm, I intended to merge this into the swift-3.0-preview-2 branch, but if this should be merged straight into master, I can re-convert the changes from master.

**Comment by abertelrud** [comment]
Oh I see.  I missed the fact that this PR is destined for `swift-3.0-preview-2-branch`, not `master`.  In that case, my comment doesn't apply.

**Pull request closed without merging**